### DISCUSSION TITLE
Missing an 'already uploaded' activity

### DISCUSSION THREAD

**Discussion by N6MIK** [philosowaffle/peloton-to-garmin#392] [Q&A] [answered]
Is there a way to re-upload an activity?  My last ride shows already uploaded in the P2G logs, but isn't appearing in Connect. 
I'm running p2g in a docker container on a Synology NAS. Thank you!

**Comment by philosowaffle** [comment, marked as answer]
The most recent version of P2G should always attempt to re-upload to Garmin.  If you're on the latest version and still seeing log messages indicating activity already uploaded, that means that Garmin itself rejected the workout as a duplicate.  You can try expanding your search on Connect to check earlier days for the workout, if the workout was not being uploaded with the correct timestamp it could appear on a previous day.

**Comment by N6MIK** [comment]
Thank you. I did some digging and found that that activity uploaded by my watch was the 'dupe' - I deleted it and the bike activity uploaded no problem. This lead me to another issue, which I'll post about in another thread.
